aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -460,12 +460,23 @@ static void nan_to_num_kernel(
 }
 
 static void kaiser_window_kernel(TensorIteratorBase& iter, int64_t window_length, double beta){
-  AT_DISPATCH_FLOATING_TYPES_AND(kBFloat16, iter.dtype(), "kaiser_window_cpu", [&](){
-    const scalar_t alpha = static_cast<scalar_t>((window_length - 1) / 2.0);
-    cpu_kernel(iter, [=](scalar_t a){
-        return calc_i0(static_cast<scalar_t>(beta) * std::sqrt(1 - std::pow((a - alpha) / alpha, static_cast<scalar_t>(2.0)))) / calc_i0(static_cast<scalar_t>(beta));
+  if (at::isReducedFloatingType(iter.dtype())) {
+    AT_DISPATCH_REDUCED_FLOATING_TYPES(iter.dtype(), "kaiser_window_cpu", [&]() {
+      const float alpha = static_cast<float>((window_length - 1) / 2.0);
+      const float beta_ = static_cast<float>(beta);
+      cpu_kernel(iter, [=](scalar_t a){
+          return calc_i0(beta_ * std::sqrt(1 - std::pow((static_cast<float>(a) - alpha) / alpha, static_cast<float>(2.0)))) / calc_i0(beta_);
+      });
     });
```

Comment:
Can we combine the implementation with float and use the `opmath_t<scalar_t>` for intermediate dtype?